app/scripts/modules/core/src/pipeline/config/graph/PipelineGraph.tsx

```diff
@@ -1,11 +1,464 @@
+import * as React from 'react';
+import * as DOMPurify from 'dompurify';
+import * as $ from 'jquery';
+import * as classNames from 'classnames';
+import autoBindMethods from 'class-autobind-decorator';
+import { clone, find, flatten, forOwn, groupBy, max, maxBy, sortBy, sum, sumBy, uniq } from 'lodash';
+import { Throttle } from 'lodash-decorators';
+import { Subscription } from 'rxjs';
+
 import { IExecution, IPipeline } from 'core/domain';
-import { IExecutionViewState } from './pipelineGraph.service';
-import { IPipelineNode } from './pipelineGraph.service';
+import { IPipelineValidationResults } from 'core/pipeline/config/validation/pipelineConfig.validator';
+import { ReactInjector } from 'core/reactShims';
+import { UUIDGenerator } from 'core/utils/uuid.service';
+
+import { IExecutionViewState, IPipelineGraphLink, IPipelineGraphNode, PipelineGraphService } from './pipelineGraph.service';
+import { PipelineGraphLink } from './PipelineGraphLink';
+import { PipelineGraphNode } from './PipelineGraphNode';
+
+import './pipelineGraph.less'
 
 export interface IPipelineGraphProps {
+  execution?: IExecution;
+  onNodeClick: (node: IPipelineGraphNode) => void;
   pipeline?: IPipeline;
-  execution: IExecution;
   viewState: IExecutionViewState;
-  onNodeClick: (node: IPipelineNode) => void;
-  shouldValidate?: boolean;
```

Comment:
you sure you want to get rid of this?

Reply:
Nope, stupid string search for `shouldValidate` not thinking about angular `should-validate`.